src/main/java/elberger/maze/Maze.java

```diff
@@ -4,145 +4,147 @@
 
 public class Maze
 {
-	private int x;
-	private int y;
-	private Cell cell;
-	private String maze[][];
-	private LinkedList<Cell> visited = new LinkedList<Cell>();
-	private LinkedList<Cell> neighbors = new LinkedList<Cell>();
-	private Stack stack = new Stack();
-
-	public Maze(int x, int y)
-	{
-		this.x = x;
-		this.y = y;
-		maze = new String[x][y];
-	}
+	private int width;
+	private int height;
+	private Cell maze[][];
 
-	public String[][] fillAllMazeCells()
+	public Maze(int width, int height)
 	{
-		for (int x = 0; x < this.x; x++)
+		this.width = width;
+		this.height = height;
+		maze = new Cell[width][height];
+
+		for (int i = 0; i < width; i++)
 		{
-			for (int y = 0; y < this.y; y++)
+			for (int j = 0; j < height; j++)
 			{
-				maze[x][y] = "_";
+				maze[i][j] = new Cell(i, j);
 			}
 		}
-		return maze;
 	}
 
-	public int getX()
+	public int getWidth()
 	{
-		return x;
+		return width;
 	}
 
-	public int getY()
+	public int getHeight()
 	{
-		return y;
+		return height;
 	}
 
-	public LinkedList<Cell> getNeighbors(Cell cell)
+	public void searchPath(Cell cell)
 	{
-		if (cell.getX() == 0)
+		Stack<Cell> stack = new Stack<>();
+		stack.push(cell);
+		cell.setVisited(true);
+
+		while (!stack.isEmpty())
 		{
-			Cell eastNeighbor = new Cell(cell.getX(), cell.getY() - 1);
-			Cell westNeighbor = new Cell(cell.getX(), cell.getY() + 1);
-			Cell southNeighbor = new Cell(cell.getX() + 1, cell.getY());
+			Cell currentCell = stack.pop();
+			Cell nextCell = getNeighbors(currentCell).get(0);
 
-			neighbors.push(eastNeighbor);
-			neighbors.push(westNeighbor);
-			neighbors.push(southNeighbor);
+			if (!nextCell.isVisited())
+			{
+				nextCell.setVisited(true);
+				breakWalls(currentCell, nextCell);
+				stack.push(nextCell);
+			} else
+			{
+				stack.pop();
```

Comment:
if this neighbor is visited then want to check all the other neighbors before popping the stack